From b3a9f8a9cd4bf4113d6477ae6e6ed74d4667707b Mon Sep 17 00:00:00 2001
From: Harsh Modi <hmodi@aviatrix.com>
Date: Fri, 24 Nov 2023 12:16:03 -0800
Subject: [PATCH] Allow substitution for the message type in type attributes

---
 src/code_generator.rs | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/code_generator.rs b/src/code_generator.rs
index 2a4d241..e7ea3c2 100644
--- a/src/code_generator.rs
+++ b/src/code_generator.rs
@@ -266,7 +266,11 @@ impl<'a> CodeGenerator<'a> {
         assert_eq!(b'.', fq_message_name.as_bytes()[0]);
         for attribute in self.config.type_attributes.get(fq_message_name) {
             push_indent(self.buf, self.depth);
-            self.buf.push_str(attribute);
+            if attribute.contains("{MESSAGE}") {
+                self.buf.push_str(&attribute.replace("{MESSAGE}", fq_message_name));
+            } else {
+                self.buf.push_str(attribute);
+            }
             self.buf.push('\n');
         }
     }
-- 
2.39.2 (Apple Git-143)

