diff --git a/src/lib.rs b/src/lib.rs
index 178abcf..96af0b1 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -754,12 +754,23 @@ fn parse_system<B: BufRead>(
     r: &mut XMLReader<B>,
 ) -> Result<Option<String>, Error> {
     let mut buf = Vec::new();
-    let mut res = None;
+    let mut res: Option<String> = None;
     loop {
         match r.read_event_into(&mut buf) {
             Ok(XMLEvent::End(ref e)) if e.name() == orig.name() => break,
             Ok(XMLEvent::Text(e)) => {
-                res = Some(e.unescape()?.to_string());
+                println!("{:#?}", e);
+                match res.as_mut() {
+                    Some(res) => res.push_str(&e.unescape()?.to_string()),
+                    None => res = Some(e.unescape()?.to_string()),
+                }
+            }
+            Ok(XMLEvent::CData(e)) => {
+                println!("{:#?}", e);
+                match res.as_mut() {
+                    Some(res) => res.push_str(&String::from_utf8_lossy(&e)),
+                    None => res = Some(String::from_utf8_lossy(&e).to_string()),
+                }
             }
             Ok(XMLEvent::Eof) => {
                 return Err(XMLError::UnexpectedEof(format!("{:?}", orig.name())).into());
